### DISCUSSION TITLE
Overscan and borders

### DISCUSSION THREAD

**Discussion by anikom15** [86Box/86Box#3353] [General]
This thread is for collecting information on timing and borders for the various video card standards. What standards dictate and what the cards actually did could differ.

The first question is how should the borders behave? Do we want to display the entire border, or just part of it? On an actual screen, the border amount varies. By the 90s it started to become more common to have controls on the monitor to adjust H-Size and V-Size so the user could display as much or as little of the borders as they wanted to (but even before this, you could tune the size by opening up the monitor). Later cards seemed to drop support for VGA border color entirely.

You can see the entire non-blanking area drawn by a monitor if you raise the brightness enough. The non-blanking area will appear less black than the outer edge of the screen. You may need to adjust H-Size/V-Size to make the picture small enough to see it.

[Here](https://github.com/86Box/86Box/files/11480865/VGA_timing_information.pdf) is information on VGA's borders and timing.

This is all I was able to find on CGA's timing:
https://forums.nesdev.org/viewtopic.php?p=77213&sid=76d305af0779b74d1710eda57f2016cb
https://forum.vcfed.org/index.php?threads/help-with-cga-timings.73731/

**Comment by dbalsom** [comment]
In regards to CGA, we can take a clue from Area 5150. In certain effects, they use a resolution of 768x246 to effectively fill the screen with no visible overscan. I would be confident in saying this is probably the maximum visible resolution for CGA. You can't squeeze much more out of the CRTC before needing to hsync. The actual logical field is 912x262 but this includes clocks spent in hblank and vblank.

Several effects like "Let's Really Push the Boundaries" rely on a seamless transition between a video mode with a blue overscan color and a full-frame video mode with blue pixels drawn where the overscan was.  This lets the effect then erase the "overscan" on different sides of the screen successively. Quite a nice effect. But if you weren't showing the full overscan to begin with, there will be a discontinuity when the viewport resolution suddenly changes.

**Reply by anikom15** [reply]
I don't think the driver is handling this correctly. It doesn't bleed into the border, it just extends the resolution. I agree that rendering the entire picture area is better. Do you know if it's possible to draw beyond the border area (that is, in the area that should be the front and back porches)?

**Reply by dbalsom** [reply]
check out this scanline wrapping around the right side of the screen due to counter overflow:
https://youtu.be/IQ2UeIx1qIA?t=1641

**Reply by dbalsom** [reply]
here's another example, from Area 5150 - I knocked the effect's vsync timer out of alignment intentionally here
![image](https://github.com/86Box/86Box/assets/7229541/d46e91a3-9c07-41cf-935a-1d62f5885561)

the highlighted part is a single CRTC 'row'. What happened is that R0 was set < C0, so C0 counted all the way to 255 (2.23 scanlines).  Resulting in a single CRTC scanline across 3 separate monitor scanlines without a single hsync being sent by the card. 

This output is from rgb2hdmi.  I assume upon reaching the edge of the field without an hsync it just shrugs and returns to the left edge of the capture buffer. I am hesitant to hook up a real monitor to see how it would handle this as I like my monitors un-exploded.
